Build failure: yabai

I don’t know what has changed in "${stdenv.cc.targetPrefix}clang" that caused this problem

https://github.com/khaneliman/nixpkgs/blob/94b09ce439fd70446a02c8979284ac0ada251f95/pkgs/os-specific/darwin/yabai/default.nix#L117

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *yabai*

### Build log

```
error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'darwin-system-24.05.20231124.5a09cb4+darwin4.4b9b83d'
         whose name attribute is located at /nix/store/6h04ab83y72nm16hh9g3llgg0s296ynl-source/pkgs/stdenv/generic/make-derivation.nix:348:7

       … while evaluating attribute 'buildCommand' of derivation 'darwin-system-24.05.20231124.5a09cb4+darwin4.4b9b83d'

         at /nix/store/wki8fh5iv47w07xpw1ly5xliz6bdbh26-source/modules/system/default.nix:105:7:

          104|
          105|       buildCommand = ''
             |       ^
          106|         mkdir $out

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: builder for '/nix/store/d9954cj284jsa0dqkc0amf73g1j8zh12-yabai-6.0.0.drv' failed with exit code 2;
       last 10 log lines:
       > rm -f ./src/osax/loader
       > rm -f ./src/osax/loader
       > mkdir -p ./bin
       > xcrun clang src/manifest.m src/osax/payload_bin.c src/osax/loader_bin.c -std=c99 -Wall -g -O0 -fvisibility=hidden -mmacosx-version-min=11.0 -fno-objc-arc -arch x86_64  -F/System/Library/PrivateFrameworks -framework Carbon -framework Cocoa -framework CoreServices -framework SkyLight -o bin/yabai
       > In file included from src/manifest.m:46:
       > src/misc/service.h:115:9: error: call to undeclared function '_NSGetExecutablePath'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
       >     if (_NSGetExecutablePath(exe_path, &exe_path_size) < 0) {
       >         ^
       > 1 error generated.
       > make: *** [makefile:62: bin/yabai] Error 1
       For full logs, run 'nix log /nix/store/d9954cj284jsa0dqkc0amf73g1j8zh12-yabai-6.0.0.drv'.
```

### Additional context
work well 

```nix
final: prev: {
  yabai = prev.yabai.overrideAttrs (finalAttrs: previousAttrs: {
    postPatch =
      ''
        substituteInPlace makefile \
          --replace "BUILD_FLAGS    = -std=c99 -Wall -g -O0 -fvisibility=hidden -mmacosx-version-min=11.0 -fno-objc-arc -arch x86_64 -arch arm64" "BUILD_FLAGS    = -std=c99 -Wall -g -O0 -fvisibility=hidden -mmacosx-version-min=11.0 -fno-objc-arc -arch x86_64 -arch arm64 -Wno-implicit-function-declaration" \
          --replace "asan: BUILD_FLAGS=-std=c99 -Wall -g -O0 -fvisibility=hidden -fsanitize=address,undefined -mmacosx-version-min=11.0 -fno-objc-arc -arch x86_64 -arch arm64" "asan: BUILD_FLAGS=-std=c99 -Wall -g -O0 -fvisibility=hidden -fsanitize=address,undefined -mmacosx-version-min=11.0 -fno-objc-arc -arch x86_64 -arch arm64 -Wno-implicit-function-declaration" \
          --replace "tsan: BUILD_FLAGS=-std=c99 -Wall -g -O0 -fvisibility=hidden -fsanitize=thread,undefined -mmacosx-version-min=11.0 -fno-objc-arc -arch x86_64 -arch arm64" "tsan: BUILD_FLAGS=-std=c99 -Wall -g -O0 -fvisibility=hidden -fsanitize=thread,undefined -mmacosx-version-min=11.0 -fno-objc-arc -arch x86_64 -arch arm64 -Wno-implicit-function-declaration" \
          --replace "install: BUILD_FLAGS=-std=c99 -Wall -DNDEBUG -O2 -fvisibility=hidden -mmacosx-version-min=11.0 -fno-objc-arc -arch x86_64 -arch arm64" "install: BUILD_FLAGS=-std=c99 -Wall -DNDEBUG -O2 -fvisibility=hidden -mmacosx-version-min=11.0 -fno-objc-arc -arch x86_64 -arch arm64 -Wno-implicit-function-declaration"
      ''
      + previousAttrs.postPatch;
  });
}

```

### Notify maintainers
@khaneliman

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 23.1.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.17.1`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

--- a/pkgs/os-specific/darwin/yabai/default.nix
+++ b/pkgs/os-specific/darwin/yabai/default.nix
@@ -108,6 +108,11 @@ in
     dontConfigure = true;
     enableParallelBuilding = true;
 
+    env = {
+      # silence service.h error
+      NIX_CFLAGS_COMPILE = "-Wno-implicit-function-declaration";
+    };
+
     postPatch = ''
       # aarch64 code is compiled on all targets, which causes our Apple SDK headers to error out.
       # Since multilib doesnt work on darwin i dont know of a better way of handling this.

